packages/DynamicHyperlink/src/DynamicHyperlink.js

```diff
@@ -11,46 +11,69 @@ export default function DynamicHyperlink({ onFetch }) {
   const I18n = useI18n();
 
   React.useEffect(() => {
+    function replaceLink(dynamicHyperlink) {
+      const originalLinkUrl = dynamicHyperlink.href;
+      const attributeValue = dynamicHyperlink.getAttribute("data-dynamic-hyperlink");
+      dynamicHyperlink.innerHTML = I18n.t("dynamicHyperlink.loading"); // eslint-disable-line no-param-reassign
+      dynamicHyperlink.setAttribute("data-dynamic-hyperlink--processed", "true");
+
+      // the consumer does the fetching of data
+      const prom = onFetch(originalLinkUrl, attributeValue);
+      if (isPromise(prom)) {
+        prom
+          .then(response => response.json())
+          .then(response => {
+            const { error, name, term } = response;
+
+            const className = error ? "invalid" : "valid";
```

Comment:
who owns this invalid/valid className (the `styling`)? 

shouldn't these styles be declared on this component? or what is the purpose for adding those?